packages/local-storage-discovery/src/index.ts

```diff
@@ -0,0 +1,154 @@
+    private readonly options?: LocalStorageDiscoveryOptions
+  ) {
+    super();
+    this.isStarted = false;
+  }
+  get [Symbol.toStringTag](): string {
+    return "@waku/local-storage-discovery";
+  }
+
+  async start(): Promise<void> {
+    if (this.isStarted) {
+      return;
+    }
+
+    log.info("Starting Local Storage Discovery");
+
+    // only listen to healthy connected peers that have identified themselves
+    this.components.events.addEventListener("peer:update", this.handleNewPeers);
+
+    // get all peers from local storage
+    const localStoragePeers = this.getPeersFromLocalStorage();
+
+    //  update the tags of the peers in the peer store & dispatch them
+    for (const { id: idStr, address } of localStoragePeers) {
+      const peerId = await createFromJSON({
+        id: idStr
+      });
+
+      await this.components.peerStore.save(peerId, {
+        tags: {
+          [DEFAULT_PEER_EXCHANGE_TAG_NAME]: {
+            value: this.options?.tagValue ?? DEFAULT_PEER_EXCHANGE_TAG_VALUE,
+            ttl: this.options?.tagTTL ?? DEFAULT_PEER_EXCHANGE_TAG_TTL
+          }
+        }
+      });
+
+      this.dispatchEvent(
+        new CustomEvent<PeerInfo>("peer", {
+          detail: {
+            id: peerId,
+            multiaddrs: [multiaddr(address)]
+          }
+        })
+      );
+    }
+
+    log.info(`Discovered ${localStoragePeers.length} peers`);
+  }
+
+  stop(): void | Promise<void> {
+    if (!this.isStarted) return;
+    log.info("Stopping Local Storage Discovery");
+    this.components.events.removeEventListener(
+      "peer:update",
+      this.handleNewPeers
+    );
+    this.isStarted = false;
+  }
+
+  handleNewPeers = (event: CustomEvent<PeerUpdate>): void => {
+    const { peer } = event.detail;
+
+    // Extract the websocket ma
+    const websocketMultiaddr = peer.addresses.find((addr) =>
+      addr.toString().includes("ws" || "wss")
+    );
+    if (!websocketMultiaddr) return;
+
+    // Retrieve peers from local storage
+    const storedPeersData = localStorage.getItem("waku:peers");
+    const localStoragePeers = (
+      storedPeersData ? JSON.parse(storedPeersData) : []
+    ) as LocalStoragePeerInfo[];
+
+    // Find if the peer already exists
+    const existingPeerIndex = localStoragePeers.findIndex(
+      (_peer) => _peer.id === peer.id.toString()
+    );
+
+    if (existingPeerIndex >= 0) {
+      // Update existing peer's address
+      localStoragePeers[existingPeerIndex].address =
+        websocketMultiaddr.toString();
+    } else {
+      // Add new peer
+      localStoragePeers.push({
+        id: peer.id.toString(),
+        address: websocketMultiaddr.toString()
+      });
+    }
+
+    this.setPeersInLocalStorage(localStoragePeers);
+  };
+
+  private getPeersFromLocalStorage(): LocalStoragePeerInfo[] {
+    const storedPeersData = localStorage.getItem("waku:peers");
+    const localStoragePeers = (
+      storedPeersData ? JSON.parse(storedPeersData) : []
```

Comment:
There are couple of things to improve.

`JSON.parse` throws so let's rewrite it to something like:
```ts
try {
  const peers = JSON.parse(data);
} catch (err) {
 return [];
}
```

Another thing is - it is not guaranteed that the string from local storage will have necessary structure. In order to guard against it we should use similar approach as I did here:
- generator - https://github.com/waku-org/js-rln/blob/master/scripts/schema_validator_codegen.js
- validator - https://github.com/waku-org/js-rln/blob/master/src/keystore/schema_validator.ts

This way after parsing the string we can check for necessary structure in run time.

I propose to split it out as a follow up task and instead rewrite this method to:
```ts
private getPeersFromLocalStorage(): any[] {
```

Then in 
```ts
for (const { id: idStr, address } of localStoragePeers) {
```
before accessing a peer - check if it has `id` and `address` properties.